### PR TITLE
Add roadmap.

### PR DESCRIPTION
**Pull request by concretevitamin**
To ease and align community contributions, this PR adds a roadmap doc, starting with directions for the next major release (v0.3).

The directions and detailed items are not meant to be totally comprehensive. That said, they cover the most actively discussed/worked-on items.

### CODE REVIEW

**Review by romilbhardwaj — APPROVED** [review]
lgtm! We can move to better tools for roadmap tracking/management later as needed

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# SkyPilot Roadmap
+
+## v0.3
```
Comment (Michaelvll) [inline ROADMAP.md:3]: I would like to call this `v1.0.0-dev0` or `v1.0.0` to align with the current master version. Reason: I believe the `v0.3` will not include all the features listed here and we should not make a promise for these features to be included in `v0.3`. This doc is a longer-term plan than `v0.3`.

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+### Managed Spot
+- Minimize the cost of the controller
+  - Support spot controller on existing cluster (or local cluster)
+  - Reducing the fixed cost of the controller (e.g., allow setting controller VM type)
+  - Increasing parallelism (number of concurrent jobs)
+- Pushing the scale (e.g., support a high number of pending/concurrent jobs)
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply ROADMAP.md:10]: +1. IIUC, this should refer to scaling up the controller itself. Probably, we can change it to `Scaling up controller and fault tolerance`?

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+### Programmatic API
+- Refactor/extend the current API to *make it easy to programmatically use SkyPilot*
+- Expose core classes in docs
+
+### More clouds
+- Refactoring of interfaces to ease adding new clouds
+- IBM Cloud
+
+### On-prem
+- Design for switching between cloud and on-prem
```
Comment (Michaelvll) [inline ROADMAP.md:29]: Do we want to add an item before this: `Robustify the on-prem feature`?

**Review by Michaelvll — APPROVED** [review]
Thanks for adding this @concretevitamin! Left serveral comments. : )

**Comment by concretevitamin** [comment]
Thanks @romilbhardwaj @Michaelvll. All addressed. Merging.

**Pull request merged**